### ISSUE TITLE
Spacemouse Enterprise not recognized

### ISSUE THREAD

**Issue by WHWFD** [smartavionics/RawMouse#1]
RawMouse seems to be loaded and is listed under Extensions, but Show Device Information is empty.
Cura.log shows, that no HID device was found.

[cura.log](https://github.com/smartavionics/RawMouse/files/4305605/cura.log)

**Comment by smartavionics** [comment]
Please edit the config.json file and add this line into the list of devices:

    [ "0x256f", "0xc633", "spacemouse", "3Dconnexion Space Mouse Enterprise" ],

JSON is very fussy about commas so if you put it at the end of the list you will need to add a comma to the preceding line and remove the comma at the end of this new line.

With this added, it should find the device.

**Comment by WHWFD** [comment]
Spacemouse Enterprise is active now, and Device Information shows up.
I'd like to change directions for the knob response, but perhaps I can do it myself.

Thanks
Walter

**Comment by DivingDuck** [comment]
You can invert this by changing the sign for scale e.g. :

{ "offset": 0.0, "scale": -0.1, "threshold": 0.01, "target": "movx" }
to
{ "offset": 0.0, "scale": 0.1, "threshold": 0.01, "target": "movx" }

**Comment by smartavionics** [comment]
That's right, thanks for the help @DivingDuck .

**Comment by WHWFD** [comment]
Meanwhile I've made some direction changes for the knob, similar to my settings in Inventor. See attached config.json.
It seems to me, that pitch and roll rotations are centered to a pivot point at the top of the working box. Can this be set-up different, maybe to the origin point?
Attached is a listing of available buttons for the SM Enterprise. In Inventor, I'm only using a few of them. It would be nice, to configure them, but I don't know their internal names.


[config.zip](https://github.com/smartavionics/RawMouse/files/4306503/config.zip)
![SM Enterprise buttons](https://user-images.githubusercontent.com/13252434/76212225-4b407f80-6208-11ea-9e43-2849c968a8eb.jpg)

**Comment by smartavionics** [comment]
Yes, the rotation is weird because it's still using the same rotation code as used for the 2D mouse with the addition of the roll axis. I have tinkered with the rotation but don't really have the time or motivation to radically change how it works.

It is certainly possible to add button functions but as I don't have any need for that I will not be putting time into that either.

Of course, pull requests are always welcome if you care to add some extra button functions I would certainly consider using that.
